### PR TITLE
MRG: Present is the __future__

### PR DESCRIPTION
**Pull request by larsoner**
Transitioning 0.18 to eradicate 2.7 support.

- [x] Remove `__future__` statements
- [x] Tweak `.travis.yml` to use 3.5+ (~~I also tried to tell it to use 3.7, this might not work, never tried~~)
- [x] Bump min requirements to what can actually be used with 3.5+
- [x] Remove things from `fixes.py` that are no longer necessary with newer deps
- [x] Remove `mne/externals/six.py` and all references
- [x] Update `whats_new.rst` to mention our change to 3k only
- [x] Make CIs happy

This should hopefully simplify our code quite a bit.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -78,18 +78,18 @@ Dependencies
...
 
-- Matplotlib >= 1.3
+- Matplotlib >= 1.5
 - Mayavi >= 4.6
 - PySurfer >= 0.8
-- Scikit-learn >= 0.15 (>= 0.18 recommended)
+- Scikit-learn >= 0.18
 - NiBabel >= 2.1.0
-- Pandas >= 0.13
+- Pandas >= 0.18
```
Comment (larsoner) [inline README.rst]: These newer dependencies reflect what I actually had to use to get a working Python 3.5 env on Travis with `conda`. I think they are reasonable numbers at this point anyway

Comment (massich) [reply README.rst]: +1

**Comment by massich** [comment]
Travis red, I'm fixing it but I cannot push in your branch.

**Comment by larsoner** [comment]
> Travis red, I'm fixing it but I cannot push in your branch.

I have `allow edits from maintainers` checked, are you sure you have the correct write-capable url (e.g., `git@github.com:/` instead of `git://github.com/`) for me?

**Comment by massich** [comment]
```sh
(mne) ❯ git remote -v                                                   
agramfort	git@github.com:agramfort/mne-python.git (fetch)
agramfort	git@github.com:agramfort/mne-python.git (push)
larnoser	git@github.com:larsoner/mne-python.git (fetch)
larnoser	git@github.com:larsoner/mne-python.git (push)
massich	git@github.com:massich/mne-python.git (fetch)
massich	git@github.com:massich/mne-python.git (push)
mikolaj	git@github.com:mmagnuski/mne-python.git (fetch)
mikolaj	git@github.com:mmagnuski/mne-python.git (push)
mne	git@github.com:mne-tools/mne-python.git (fetch)
mne	git@github.com:mne-tools/mne-python.git (push)
slasnista	git@github.com:Slasnista/mne-python.git (fetch)
slasnista	git@github.com:Slasnista/mne-python.git (push)
stefan_appelhoff	git@github.com:sappelhoff/mne-python.git (fetch)
stefan_appelhoff	git@github.com:sappelhoff/mne-python.git (push)
```
Usually I can push with this config

**Comment by massich** [comment]
Now I could without a problem. I've no idea!

**Comment by larsoner** [comment]
`larnoser` is a typo, should be `larsoner`

**Comment by massich** [comment]
The typo is only on my remote name, the url is fine. but I'll fix my typo anyway.

**Comment by larsoner** [comment]
I'm suggesting that you might have tried to `git push larsoner` before but it worked this time because you did `git push larnoser`

**Comment by larsoner** [comment]
`make flake` reveals some remaining name errors

**Comment by massich** [comment]
> make flake reveals some remaining name errors

It does not for me:
```
~/code/mne-python py3k* ⇣⇡
(mne) ❯ make flake                  
Running flake8
0
flake8 passed
```

**Comment by massich** [comment]
I'm not sure that the encoding is properly done. I managed to make it green, but I'm not sure that I did the right thing.

**Comment by larsoner** [comment]
Okay I'll take a look and push if necessary

**Comment by massich** [comment]
I've restarted some jobs. 

BTW, our CI dropped to 12 min ??

**Comment by larsoner** [comment]
It's because our testing dataset is not downloading properly (most tests skipped), fix forthcoming hopefully

**Comment by larsoner** [comment]
Ready for review/merge from my end. CIs should come back happy but if they don't, I'll tweak a bit more.

This should go in before we take care of deprecations, which I plan to do next.

**Comment by agramfort** [comment]
good to go form my side if CIs are green

**Comment by massich** [comment]
those are suspects:
`pytest mne/beamformer/tests/test_rap_music.py mne/decoding/tests/test_receptive_field.py mne/decoding/tests/test_search_light.py mne/inverse_sparse/tests/test_gamma_map.py mne/inverse_sparse/tests/test_mxne_inverse.py mne/io/edf/tests/test_edf.py mne/simulation/tests/test_evoked.py mne/tests/test_annotations.py mne/viz/tests/test_ica.py mne/viz/tests/test_raw.py
`

**Comment by larsoner** [comment]
Yeah I am fixing them currently

**Comment by massich** [comment]
cool, most of them are userwarnings or runtimewarnings raised by these old versions. Maybe we can require a new version in the testing or add the logic necessary to make them compatible. (I'll check your commit, and keep it in mind for the next time)

**Comment by larsoner** [comment]
I'm just adding to `setup.cfg` for now if they are non-critical warnings

File: setup.cfg [hunk 2]
```diff
@@ -32,6 +32,9 @@ filterwarnings =
     ignore:Importing from numpy.testing.decorators is deprecated, import from numpy.testing instead:DeprecationWarning
     ignore:np.loads is deprecated, use pickle.loads instead:DeprecationWarning
     ignore:The oldnumeric module will be dropped:DeprecationWarning
+    ignore:Collection picker None could not be converted to float:UserWarning
+    ignore:covariance is not positive-semidefinite:RuntimeWarning
+    ignore:Can only plot ICA components:RuntimeWarning
```
Comment (massich) [inline setup.cfg]: the matplotlib warning is missing 
`Matplotlib is building the font cache using fc-list`

I'm not sure if we should ignore it here or require a version, like with `sklearn`. Is there a way to change the setup depending on the versions?

Comment (massich) [reply setup.cfg]: maybe some fancy `@pytest.mark.filterwarnings("ignore:api v1")` depending on the version. But I'm not sure if its worth it. On the other hand, ignore them always might not be a good strategy either.

**Pull request merged**